lib/extensions/descendant_loader.rb

```diff
@@ -261,20 +261,73 @@ def scoped_name(name, scopes)
   end
 
   module ArDescendantsWithLoader
+    # Use caution if modifying the list of excluded caller_locations below.
+    # These methods are called very early from rails during code load time,
+    # before the models are even loaded, as documented below.
+    # Ideally, a more resilient conditional can be used in the future.
+    #
+    # Called from __update_callbacks during model load time:
+    #   "xxx/gems/activesupport-7.0.8.4/lib/active_support/callbacks.rb:706:in `__update_callbacks'",
+    #   "xxx/gems/activesupport-7.0.8.4/lib/active_support/callbacks.rb:764:in `set_callback'",
+    #   "xxx/gems/activemodel-7.0.8.4/lib/active_model/validations.rb:172:in `validate'",
+    #   "xxx/gems/activemodel-7.0.8.4/lib/active_model/validations/with.rb:96:in `block in validates_with'",
+    #   "xxx/gems/activemodel-7.0.8.4/lib/active_model/validations/with.rb:85:in `each'",
+    #   "xxx/gems/activemodel-7.0.8.4/lib/active_model/validations/with.rb:85:in `validates_with'",
+    #   "xxx/gems/activemodel-7.0.8.4/lib/active_model/validations/format.rb:109:in `validates_format_of'",
+    #   "xxx/gems/ancestry-4.1.0/lib/ancestry/has_ancestry.rb:30:in `has_ancestry'",
+    #   "xxx/manageiq/app/models/vm_or_template.rb:15:in `<class:VmOrTemplate>'",
+    #   "xxx/manageiq/app/models/vm_or_template.rb:6:in `<main>'",
     def descendants
-      if Vmdb::Application.instance.initialized? && !defined? @loaded_descendants
+      if Vmdb::Application.instance.initialized? && !defined?(@loaded_descendants) && %w[__update_callbacks].exclude?(caller_locations(1..1).first.base_label)
```

Comment:
the bot/cop complained about using caller_locations to build the full array so I accepted that suggestion even though I think it makes it a bit more convoluted. This method can be run many times, especially when models aren't yet loaded, so it should be as performant as possible.